Automattic/jetpack
# theme-tools: remove jQuery from responsive-videos

This PR reimplements the `responsive-videos.js` script in vanilla JS, removing the dependency on `jQuery`.

## Proposed changes:
* Reimplement `responsive-videos.js` without `jQuery`
* Remove the dependency on `jQuery` for this script
* Add `responsive-videos` to list of paths to be compiled
* Modify `moduleSources` handling to support sub-paths

### Other information:

- N/A Have you written new tests for your changes, if applicable?

## Jetpack product discussion
None; this is a simple under-the-hood change.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
- For your test blog, pick a theme that doesn't require `jQuery` by itself but makes use of `responsive-videos.js`, e.g. Independent Publisher 2.
- Create a test page with a video shortcode.
   <details>
   <summary>Example content</summary>

   ```html
   
   <p class="has-text-align-center">A centered paragraph.</p>
   
   
   
   [youtube https://www.youtube.com/watch?v=dQw4w9WgXcQ]
   
   ```

   </details>
- Load the test page.
- Ensure that `jQuery` is not loaded.
- Ensure that the video resizes as you shrink and grow the window, and that it does so in a manner identical to the current script in trunk.
